neqo-bin/src/server/mod.rs

```diff
@@ -224,36 +225,44 @@ impl ServerRunner {
         }
     }
 
-    /// Tries to find a socket, but then just falls back to sending from the first.
-    fn find_socket(&mut self, addr: SocketAddr) -> &mut crate::udp::Socket {
-        let ((_host, first_socket), rest) = self.sockets.split_first_mut().unwrap();
-        rest.iter_mut()
-            .map(|(_host, socket)| socket)
-            .find(|socket| {
-                socket
-                    .local_addr()
-                    .ok()
-                    .map_or(false, |socket_addr| socket_addr == addr)
-            })
-            .unwrap_or(first_socket)
-    }
```

Comment:
Moved outside of `impl ServerRunner {}` as it no longer takes `&mut self` but instead `socket: &mut [(SocketAddr, crate::udp::Socket)]`.

Reply:
You can move that back into `ServerRunner`.  Because it is dealing with internals of that struct, it might make sense to do that.  The only cost is that you invoke with `Self::process` rather than just `process`.

Reply:
Sounds good. Done in https://github.com/mozilla/neqo/pull/2202/commits/1b7ce8fb01cf67b6a65e74d8e654bf5aea02a028.